ch_tools/chadmin/cli/table_metadata.py

```diff
@@ -0,0 +1,93 @@
+import re
+import uuid
+
+UUID_PATTERN = "UUID"
+OFFSET_UUID_PATTERN = 1
+ENGINE_PATTERN = "ENGINE"
+
+PARSER_SUPPORT_TABLE_ENGINE = [
+    "MergeTree",
+    "ReplacingMergeTree",
+    "SummingMergeTree",
+    "AggregatingMergeTree",
+    "CollapsingMergeTree",
+    "VersionedCollapsingMergeTree",
+    "ReplicatedMergeTree",
+]
+
+
+class TableMetadata:
+    @property
+    def table_uuid(self):
+        return self._table_uuid
+
+    @property
+    def table_engine(self):
+        return self._table_engine
+
+    def __init__(self, table_metadata_path: str):
+        assert table_metadata_path.endswith(".sql")
+
+        self._table_uuid = None
+        self._table_engine = None
+
+        with open(table_metadata_path, "r", encoding="utf-8") as metadata_file:
+            for line in metadata_file:
+                if UUID_PATTERN in line:
+                    self._table_uuid = TableMetadata._parse_uuid(line)
+                    continue
+                if ENGINE_PATTERN in line:
+                    result = TableMetadata._parse_engine(line)
+                    self._table_engine = result
+                    break
+
+        if self._table_uuid is None:
+            raise RuntimeError(f"Empty UUID from metadata: '{table_metadata_path}'")
+
+        if self._table_engine is None:
+            raise RuntimeError(
+                f"Empty table engine from metadata: '{table_metadata_path}'"
+            )
+
+    @staticmethod
+    def _is_valid_uuid(uuid_str: str) -> bool:
```

Comment:
I can't imagine the case when we will need it. Lets make it non-static by now

Reply:
It is a way to check that metadata doesn't broken. Its important because all next removing basing on UUID from metadata file. We use it only inside the class and so it makes sense yo use as static.

Reply:
> Why if the method is used only inside the class, then it should be static?

We don't use self so we dont have any reasons to make method nonstatic.